App/src/test/java/vut/fit/gja2023/app/CsvReaderTests.java

```diff
@@ -0,0 +1,152 @@
+        
+        assertThrows(IOException.class, () -> {
+            csvReaderService.readCsv("nonexistent/file/path", config);
+        });
+    }
+    
+    @Test
+    void testOrderMaxOutOfBounds() {
+        CsvReaderConfig config = new CsvReaderConfig(new int[]{0, 2}, true);
+        
+        assertThrows(IndexOutOfBoundsException.class, () -> {
+            csvReaderService.readCsv(filePath1, config);
+        });
+    }
+    
+    @Test
+    void testOrderMinOutOfBounds() {
+        CsvReaderConfig config = new CsvReaderConfig(new int[]{1, -1}, true);
+        
+        assertThrows(IndexOutOfBoundsException.class, () -> {
+            csvReaderService.readCsv(filePath1, config);
+        });
+    }
+    
+    @Test
+    void testOrderTooManyColumnsSpecified() {
+        CsvReaderConfig config = new CsvReaderConfig(new int[]{1, 0, 1}, true);
+        
+        assertThrows(IllegalArgumentException.class, () -> {
+            csvReaderService.readCsv(filePath1, config);
+        });
+    }
+    
+    @Test
+    void testReadExistingFileAndCheckResult() {
+        CsvReaderConfig config = new CsvReaderConfig(new int[]{0, 1}, true);
+        
+        try {
+            List<String[]> res = csvReaderService.readCsv(filePath1, config);
+            
+            for (int i = 0; i < res.size(); i++) {
+                for (int j = 0; j < TEAMS_RESULT.getFirst().length; j++) {
+                    assertEquals(TEAMS_RESULT.get(i)[j], res.get(i)[j]);
+                }
+            }
+        }
+        catch (Exception e) {
+            Logger.getLogger(CsvReaderTests.class.getName()).log(Level.SEVERE, null, e);
+        }
+    }
+    
+    @Test
+    void testReadExistingFileAndCheckResultNoHeader() {
+        CsvReaderConfig config = new CsvReaderConfig(new int[]{0, 1}, false);
+        
+        try {
+            List<String[]> res = csvReaderService.readCsv(filePath2, config);
+            
+            for (int i = 0; i < res.size(); i++) {
+                for (int j = 0; j < TEAMS_RESULT.getFirst().length; j++) {
+                    assertEquals(TEAMS_RESULT.get(i)[j], res.get(i)[j]);
+                }
+            }
+        }
+        catch (Exception e) {
+            Logger.getLogger(CsvReaderTests.class.getName()).log(Level.SEVERE, null, e);
+        }
+    }
+    
+    @Test
+    void testReadExistingFileAndCheckResultReverseOrder() {
+        CsvReaderConfig config = new CsvReaderConfig(new int[]{1, 0}, true);
+        
+        try {
+            List<String[]> res = csvReaderService.readCsv(filePath1, config);
+            
+            for (int i = 0; i < res.size(); i++) {
+                for (int j = 0; j < TEAMS_RESULT_REVERSE.getFirst().length; j++) {
+                    assertEquals(TEAMS_RESULT_REVERSE.get(i)[j], res.get(i)[j]);
+                }
+            }
+        }
+        catch (Exception e) {
+            Logger.getLogger(CsvReaderTests.class.getName()).log(Level.SEVERE, null, e);
```

Comment:
pri testoch je zbytocne logovat, takto by to zas len odchytilo nejaku chybu, zalogovalo, a test by presiel... takže daj preč try/catch, a do hlavicky funkcie throws Exception